onnxruntime/core/providers/cpu/tensor/reshape_helper.h

```diff
@@ -11,6 +11,10 @@ namespace onnxruntime {
 class ReshapeHelper {
  public:
   ReshapeHelper(const TensorShape& input_shape, TensorShapeVector& requested_shape, bool allow_zero = false) {
+    const auto input_shape_size = input_shape.Size();
```

Comment:
nit: thought `.Size()` is the product of input_shape elements, so didn't quite follow at first about just checking `input_shape_size` != -1 equates to ensure that there's no -1 dimensions in input_shape.

maybe might be good to annotate that the .Size() returns -1 when found a negative element in tensorshape?